flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2794,7 +2802,23 @@ class ScalarExprLowering {
         // address without making a copy.
         argAddr = readIfBoxValue(argAddr);
 
-        if (arg.passBy == PassBy::BaseAddress) {
+        if (arg.passBy == PassBy::BaseAddressValueAttribute) {
+          mlir::Value temp;
+          if (isArray(*expr)) {
+            auto val = genBoxArg(*expr);
```

Comment:
Is the array copy really needed if `actualArgIsVariable` is `false` for arrays ?

Reply:
> Thanks for improving the `VALUE` attribute support here. Your PR makes me realize that lowering is not covering the the assumed shape with VALUE attribute case. We should have a proper TODO in CallInterface.cpp for this case [here](https://github.com/flang-compiler/f18-llvm-project/blob/dc8541eaca6c7ea1eea945558b19043bb64fa8a5/flang/lib/Lower/CallInterface.cpp#L839) when `isValueAttr` is `true`.

I'll take a look at that in a bit - I'm away for a fair few days in the next three weeks, so not sure I can get much done. Adding a TODO may fall within what I can achieve tho'. ;)

I have removed the copy if we already have a copy - at least for the test case I made right now.